core-bundle/src/Image/Studio/ImageResult.php

```diff
@@ -64,24 +70,51 @@ class ImageResult
      *
      * @internal Use the Contao\Image\Studio\Studio factory to get an instance of this class
      */
-    public function __construct(ContainerInterface $locator, string $projectDir, $filePathOrImage, $sizeConfiguration = null)
+    public function __construct(ContainerInterface $locator, string $projectDir, $filePathOrImage, $sizeConfiguration = null, ResizeOptions $resizeOptions = null)
     {
         $this->locator = $locator;
         $this->projectDir = $projectDir;
         $this->filePathOrImageInterface = $filePathOrImage;
         $this->sizeConfiguration = $sizeConfiguration;
+        $this->resizeOptions = $resizeOptions;
     }
 
     /**
      * Creates a picture with the defined size configuration.
      */
     public function getPicture(): PictureInterface
     {
-        if (null === $this->picture) {
-            $this->picture = $this->pictureFactory()->create($this->filePathOrImageInterface, $this->sizeConfiguration);
+        if (null !== $this->picture) {
+            return $this->picture;
+        }
+
+        // Unlike the Contao\Image\PictureFactory the PictureFactoryInterface
+        // does not know about ResizeOptions. We therefore check if the third
+        // argument of the 'create' method allows setting them.
+        $canHandleResizeOptions = static function (PictureFactoryInterface $factory): bool {
+            $createParameters = (new \ReflectionClass($factory))
```

Comment:
What happend to 0d1e22cd76c486e3cf5bb7df5a089ccda1f8f851 ?

Reply:
:scream: Reapplied in ad2d259f5cc0832ddb8de98e888eb197cd8f39c4.